lib/frame.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class Frame
```

Comment:
Great idea to have frames as its own class! Very clear and concise methods, good use of encapsulation and SRP.